server/handler/frontend.go

```diff
@@ -55,6 +61,164 @@ func LoadTemplates(c *FilesConfig, basePath string) (templatetree.HTMLTree, erro
 
 			return r
 		},
+		"getPredicateInfo": func(predicateInfo *common.PredicateInfo) []string {
+			switch predicateInfo.Name {
+			case "Author":
+				return []string{predicateInfo.ContributorInfo.Author}
+			case "Contributors":
+				return predicateInfo.ContributorInfo.Contributors
+			case "Target Branch":
+				return []string{predicateInfo.BranchInfo.Branch}
+			case "Source Branch":
+				return []string{predicateInfo.BranchInfo.Branch}
+			case "Changed Files":
+				if len(predicateInfo.FileInfo.ChangedFiles) <= 4 {
+					return predicateInfo.FileInfo.ChangedFiles
+				}
+				return append(predicateInfo.FileInfo.ChangedFiles[:4], "...")
+			case "Modified Lines":
+				var modificationStats []string
+				AddedLines := predicateInfo.FileInfo.AddedLines
+				DeletedLines := predicateInfo.FileInfo.DeletedLines
+				TotalModifiedLines := predicateInfo.FileInfo.TotalModifiedLines
+				if AddedLines != 0 {
+					modificationStats = append(modificationStats, "Addition : "+strconv.FormatInt(AddedLines, 10))
+				}
+				if DeletedLines != 0 {
+					modificationStats = append(modificationStats, "Deletion : "+strconv.FormatInt(DeletedLines, 10))
+				}
+				if TotalModifiedLines != 0 {
+					modificationStats = append(modificationStats, "Total : "+strconv.FormatInt(TotalModifiedLines, 10))
+				}
+				return modificationStats
+			case "Labels":
+				return predicateInfo.LabelInfo.PRLabels
+			case "Commit Hashes":
+				return predicateInfo.CommitInfo.CommitHashes
+			case "Commit Hashes and Keys":
+				var commitsAndKeys []string
+				commitsAndKeys = append(commitsAndKeys, "Commit Hashes : "+strings.Join(predicateInfo.CommitInfo.CommitHashes[:], ", "))
+				if len(predicateInfo.CommitInfo.Keys) > 0 {
+					commitsAndKeys = append(commitsAndKeys, "Keys : "+strings.Join(predicateInfo.CommitInfo.Keys[:], ", "))
+				} else {
+					commitsAndKeys = append(commitsAndKeys, "No valid keys because commit hash is not valid.")
+				}
+				return commitsAndKeys
+			case "Commit Hashes and Signers":
+				var commitsAndSigners []string
+				commitsAndSigners = append(commitsAndSigners, "Commit Hashes : "+strings.Join(predicateInfo.CommitInfo.CommitHashes[:], ", "))
+				if len(predicateInfo.CommitInfo.Signers) > 0 {
+					commitsAndSigners = append(commitsAndSigners, "Signers : "+strings.Join(predicateInfo.CommitInfo.Signers[:], ", "))
+				} else {
+					commitsAndSigners = append(commitsAndSigners, "No valid signers because commit hash is not valid.")
+				}
+				return commitsAndSigners
+			case "Status":
+				return predicateInfo.StatusInfo.Status
+			case "Title":
+				return []string{predicateInfo.TitleInfo.PRTitle}
+			default:
+				return []string{}
+			}
+		},
+		"getPredicateRequirement": func(predicateInfo *common.PredicateInfo) map[string][]string {
+			switch predicateInfo.Type {
```

Comment:
It's odd that this switches on `Type`, while the other function switches on `Name`. We should aim to only have a single field that requires a `switch` like this to make it easier to correctly add new predicates.

Reply:
now all the functions related to rendering predicate info switch on `Type`